allow traceback formatting to be overridden

digitalkaoz opened the pull request:
i used it to implement a formatter that formats the traceback as json as well. i was not sure if you liked the additional external library (https://boltons.readthedocs.io/en/latest/tbutils.html#boltons.tbutils.ExceptionInfo.to_dict) so i made it optional and only an idea on how to improve the json format 🙂

```python
from boltons import tbutils
from sanic_json_logging.formatters import JSONFormatter


class JsonTbJSONFormatter(JSONFormatter):
    def formatStack(self, stack_info: str) -> str:
        return tbutils.ExceptionInfo.from_current().to_dict()
```

i combination with #133 it would be like this:

```python
        from sanic_json_logging import LOGGING_CONFIG_DEFAULTS

        cfg = {**LOGGING_CONFIG_DEFAULTS}
        cfg["formatters"]["generic"]["class"] = "JsonTbJSONFormatter"

        setup_json_logging(app, disable_json_access_log=True, config=cfg)
```

in the end you get a nice exception traceback:

```json
{
  "timestamp": "2021-08-26T23:19:49.412293Z",
  "level": "ERROR",
  "message": "Exception occurred while handling uri: 'http://127.0.0.1:8000/'",
  "type": "exception",
  "logger": "sanic.error",
  "worker": 31915,
  "filename": "handlers.py",
  "lineno": 146,
  "traceback": {
    "exc_type": "Exception",
    "exc_msg": "foo",
    "exc_tb": {
      "frames": [
        {
          "func_name": "handle_request",
          "lineno": 770,
          "module_name": "sanic.app",
          "module_path": "/usr/local/lib/python3.9/site-packages/sanic/app.py",
          "lasti": 182,
          "line": "                    response = await response"
        },
        {
          "func_name": "root",
          "lineno": 20,
          "module_name": "api.general",
          "module_path": "/api/controllers/general.py",
          "lasti": 6,
          "line": "    raise Exception(\"foo\")"
        }
      ]
    }
  },
  "req_id": "f128370f-b949-44e7-bb94-4635bbcad486"
}
```

terrycain commented:
Looks cool :smile:

Why don't you add that subclass to a separate file in the project and update the readme to say its there and add an extras section to the pyproject.toml ([here](https://python-poetry.org/docs/pyproject/#extras) shows how to add extras to pyproject.toml), that way people could install something like `sanic-json-logging[extratb]` for more indepth tracebacks

digitalkaoz commented:
@terrycain i added the Formatter to the package, im not sure if i did the `poetry` stuff correctly. Feel free the change it accordingly. I also added some lines to the README.

terrycain commented on README.rst:
```diff
@@ -85,9 +85,71 @@ And if you log to the ``root`` logger, inside a request, it'll look like this.
 Changelog
 ---------
 
+4.0.1
```
:+1:

terrycain commented:
Nice @digitalkaoz can you give whats in master a quick test and then i'll tag it

digitalkaoz commented:
@terrycain there was a little thing missing, fixed it by #135 all works as intended now
